daffodil-lib/src/main/scala/org/apache/daffodil/xml/DaffodilConstructingLoader.scala

```diff
@@ -288,10 +288,77 @@ class DaffodilConstructingLoader private[xml] (uri: URI,
     text(pos, s)
   }
 
+  /**
+   * Same CRLF/CR => LF processing as text gets.
+   */
   override def comment(pos: Int, s: String): Comment = {
     Comment(text(pos, s).text)
   }
 
+  /**
+   * Same CRLF/CR => LF processing as text gets.
+   */
+  override def procInstr(pos: Int, target: String, txt: String) =
+    ProcInstr(target, text(pos, txt).text)
+
+  /**
+   * Override of document to make it tolerant of the start of the file
+   * being whitespace instead of a "<" character
+   */
+  override def document(): Document = {
+
+    if ('<' == ch) super.document()
+    else {
+      // File did not start with "<".
+      // We want to tolerate whitespace and comments prior to the
+      // document element.
+
+      doc = new Document()
+
```

Comment:
If the first character is "<" then their code is correct, so that is dispensed with right at the top by just calling super. 
In the case of the first character Not being "<" then the file cannot have an XML prolog (first characters must be "<?xml " exactly, and they infer charset based on what bytes this is, etc. 

So yes, there is no possibility of the file containing an XML prolog if it doesn't start with the first character being "<".

Reply:
Rewritten.